randomwallpaper@iflow.space/adapter/localFolder.js

```diff
@@ -6,18 +6,23 @@ const HistoryModule = Self.imports.history;
 
 const BaseAdapter = Self.imports.adapter.baseAdapter;
 
-const RWG_SETTINGS_SCHEMA_LOCAL_FOLDER = 'org.gnome.shell.extensions.space.iflow.randomwallpaper.sources.localFolder';
+const RWG_SETTINGS_SCHEMA_SOURCES_LOCAL_FOLDER = 'org.gnome.shell.extensions.space.iflow.randomwallpaper.sources.localFolder';
 
 var LocalFolderAdapter = class extends BaseAdapter.BaseAdapter {
-	constructor(id, wallpaperLocation) {
+	constructor(id, name, wallpaperLocation) {
 		super(wallpaperLocation);
 
+		this._sourceName = name;
+		if (this._sourceName === null || this._sourceName === "") {
+			this._sourceName = 'Local Folder';
+		}
+
 		let path = `/org/gnome/shell/extensions/space-iflow-randomwallpaper/sources/localFolder/${id}/`;
-		this._settings = new SettingsModule.Settings(RWG_SETTINGS_SCHEMA_LOCAL_FOLDER, path);
+		this._settings = new SettingsModule.Settings(RWG_SETTINGS_SCHEMA_SOURCES_LOCAL_FOLDER, path);
 	}
 
 	requestRandomImage(callback) {
-		const folder = Gio.file_new_for_path(this._settings.get('folder', 'string'));
+		const folder = Gio.File.new_for_path(this._settings.get('folder', 'string'));
```

Comment:
I actually don't know. The docs are giving an error for [file_new_for_path](https://gjs-docs.gnome.org/gio20~2.0/gio.file_new_for_path) and there's no indication that [File.new_for_path()](https://gjs-docs.gnome.org/gio20~2.0/gio.file#function-new_for_path) is a new thing.

I can't remember exactly why I changed this either. It could be that I started using eslint around this time.

Reply:
No problem, I was just curious. Most of my code is pretty old and from a time where there was basically no documentation so I guess there is a lot of outdated stuff :shrug: